src/neo4j/_async/work/result.py

```diff
@@ -242,17 +280,81 @@ def on_success(summary_metadata):
             self._metadata.update(summary_metadata)
             self._bookmark = summary_metadata.get("bookmark")
             self._database = summary_metadata.get("db", self._database)
+            await _on_summary()
+            self._handle_warnings()
 
         # This was the last page received, discard the rest
         self._connection.discard(
             n=-1,
             qid=self._qid,
             on_success=on_success,
             on_failure=on_failure,
-            on_summary=on_summary,
         )
         self._streaming = True
 
+    def _handle_warnings(self) -> None:
+        sev_filter: t.Tuple[NotificationMinimumSeverity, ...] = ()
+        if (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.WARNING
+        ):
+            sev_filter = (NotificationMinimumSeverity.WARNING,)
+        elif (
+            self._warn_notification_severity
+            == NotificationMinimumSeverity.INFORMATION
+        ):
+            sev_filter = (NotificationMinimumSeverity.INFORMATION,
+                          NotificationMinimumSeverity.WARNING)
+
+        summary = self._obtain_summary()
+        query = self._metadata.get("query")
+        for notification in summary.summary_notifications:
+            log_call = notification_log.debug
+            if notification.severity_level == NotificationSeverity.INFORMATION:
+                log_call = notification_log.info
+            elif notification.severity_level == NotificationSeverity.WARNING:
+                log_call = notification_log.warning
+            log_call(
+                "Received notification from DBMS server: %s",
+                NotificationPrinter(notification, query, one_line=True)
+            )
+
+            if notification.severity_level not in sev_filter:
+                continue
+            warning_cls: te.Type[Warning] = Neo4jWarning
+            if notification.category == NotificationCategory.DEPRECATION:
+                warning_cls = Neo4jDeprecationWarning
+            creation_frame = self._creation_frame
+            if creation_frame is False:
+                warn(warning_cls(notification), stacklevel=1)
+            else:
+                globals_ = creation_frame.frame.f_globals
+                warning_registry = globals_.get("__warningregistry__", {})
+                warn_explicit(
+                    warning_cls(notification, query),
+                    None,
+                    creation_frame.filename,
+                    creation_frame.lineno,
+                    module=globals_["__name__"],
+                    registry=warning_registry,
+                    module_globals=globals_
+                )
+
+    @property
+    def _creation_frame(self) -> t.Union[t.Literal[False], inspect.FrameInfo]:
+        if self._creation_frame_cache is not None:
+            return self._creation_frame_cache
+        if self._creation_stack is None:
```

Comment:
Could you please put one line between the ifs? It is reading like they are the same if/else chain.

Reply:
I personally don't stumble over this as if/else chains in python are if/elif chains. So you can see from the length of the keyword alone if this is continuation or not. But I can see how this might be confusing. I'll add a blank line.